### PR TITLE
Dog command + GET Util

### PR DESCRIPTION
**Pull request by RAdu2005**
Added Dog command that gets media from API and uploads to Discord
From my research, you can't upload a file to Discord without downloading/streaming locally in JDA
Added Gson to dependencies to parse response
Added a GET request Util

Removed unnecessary logger from Help command

Might want to investigate GIF media never showing up. There is no error, there just never happens to receive a .gif in API response (in my testing).

### CODE REVIEW

**Comment by 0mlml** [comment]
Hmmm, I accessed it in my browser a couple times and saw a .gif pop up. That's where I got that from.

File: build.gradle [hunk 1]
```diff
@@ -23,6 +23,7 @@ dependencies {
     annotationProcessor "org.projectlombok:lombok:$lombokVersion"
 
     implementation "ch.qos.logback:logback-classic:1.5.6"
+    implementation 'com.google.code.gson:gson:2.11.0'
```
Comment (0mlml) [inline build.gradle]: A whole dependency for this is awful, but I guess .JSON was mandatory at some point for interacting with APIs. Does JDA already have a JSON library so we don't have two?

File: src/main/java/dev/mlml/command/impl/Dog.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            return;
+        }
+
+        String mediaURl = response.get("url").getAsString();
+        logger.info("Media URL is: " + mediaURl);
+        TextChannel channel = (TextChannel) ctx.getChannel();
+
+        try {
+            URL url = new URL(mediaURl);
+            String fileName = mediaURl.substring(mediaURl.lastIndexOf('/') + 1);
```
Comment (0mlml) [inline src/main/java/dev/mlml/command/impl/Dog.java]: Isn't there a function to parse this?

File: src/main/java/dev/mlml/command/impl/Dog.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+            try (InputStream in = url.openStream();
+                 ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+
+                byte[] buffer = new byte[1024];
```
Comment (0mlml) [inline src/main/java/dev/mlml/command/impl/Dog.java]: 1024 bytes maximum cannot be correct...

**Review by 0mlml — CHANGES_REQUESTED** [review]
If a JSON library is already imported, we should probably redo the config in JSON

File: src/main/java/dev/mlml/command/impl/Dog.java (hunk 3 above)
Comment (RAdu2005) [reply src/main/java/dev/mlml/command/impl/Dog.java]: ChatGPT and it works decent even on videos

File: build.gradle (hunk 1 above)
Comment (RAdu2005) [reply build.gradle]: I'm not sure, their documentation is a bit hard to navigate.

File: src/main/java/dev/mlml/command/impl/Dog.java (hunk 2 above)
Comment (RAdu2005) [reply src/main/java/dev/mlml/command/impl/Dog.java]: where?

File: build.gradle (hunk 1 above)
Comment (0mlml) [reply build.gradle]: I was thinking you could just check their dependencies list

File: src/main/java/dev/mlml/command/impl/Dog.java (hunk 2 above)
Comment (0mlml) [reply src/main/java/dev/mlml/command/impl/Dog.java]: "ChatGPT, how do I parse a filename from a URL?" 
and see if it has a better solution

File: src/main/java/dev/mlml/command/impl/Dog.java (hunk 3 above)
Comment (0mlml) [reply src/main/java/dev/mlml/command/impl/Dog.java]: Okay

**Comment by RAdu2005** [comment]
Increased buffer to 4096.
Found a JDA Json library and used that instead
used .getFile() for name of file

File: src/main/java/dev/mlml/Utils.java [hunk 4]
```diff
@@ -12,4 +23,29 @@ public static boolean stringIsUserMention(String input) {
     public static boolean stringIsChannelMention(String input) {
         return input.matches("^<#\\d{17,19}>$");
     }
+
+    public static DataObject sendGetRequest(String urlString) {
```
Comment (0mlml) [inline src/main/java/dev/mlml/Utils.java:27]: For completeness sake, a post request should be implemented at this point, I'll make a ticket for it as it's out of scope for this PR

File: src/main/java/dev/mlml/command/impl/Dog.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+        final DataObject response = Utils.sendGetRequest(DOG_API_URL);
+
+        if (Objects.isNull(response)) {
+            ctx.getMessage().reply("Could not fetch media").queue();
+            return;
+        }
+
+        String mediaURl = response.get("url").toString();
+        logger.info("Media URL is: " + mediaURl);
```
Comment (0mlml) [inline src/main/java/dev/mlml/command/impl/Dog.java]: This should be a debug print

**Review by 0mlml — CHANGES_REQUESTED** [review]

**Review by 0mlml — APPROVED** [review]

**Pull request merged**